src/schnetpack/nn/embed.py

```diff
@@ -0,0 +1,278 @@
+"""
+Geometrical embeddings from the atomic positions and its neighbors.
+"""
+import logging
+from functools import partial
+
+import numpy as np
+
+from typing import Dict, Sequence
+
+from torch import nn
+
+from src.schnetpack import properties
+from src.schnetpack.atomistic import PairwiseDistances
+from src.schnetpack.nn.radial import get_rbf_fn
+from src.schnetpack.nn.spherical import init_sph_fn
+from src.schnetpack.nn.cutoff import get_cutoff_fn
+from src.schnetpack.utils.mask import safe_scale, safe_mask
+
+
+def add_cell_offsets(r_ij: np.ndarray, cell: np.ndarray, cell_offsets: np.ndarray):
+    """
+    Add offsets to distance vectors given a cell and cell offsets. Cell is assumed to be
+    Args:
+        r_ij (Array): Distance vectors, shape: (n_pairs,3)
+        cell (Array): Unit cell matrix, shape: (3,3). Unit cell vectors are assumed to be row-wise.
+        cell_offsets (Array): Offsets for each pairwise distance, shape: (n_pairs,3).
+    Returns:
+    """
+    offsets = np.einsum('...i, ij -> ...j', cell_offsets, cell)
+    return r_ij + offsets
+
+
+def _init_sphc_zeros(z, sph_ij, *args, **kwargs):
+    return {'chi': np.zeros((z.shape[-1], sph_ij.shape[-1]), dtype=sph_ij.dtype)}
+
+
+def _init_sphc(z, sph_ij, phi_r_cut, idx_i, point_mask, mp_normalization, *args, **kwargs):
```

Comment:
we can keep it but please rewrite so that all comments in this PR are considered (safe_scale, torch instead of numpy, point_mask)